lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -224,65 +233,61 @@ export default class BitcoinProcessor {
 
   private async processBlockData (
     blockData: {[blockHash: string]: BitcoinBlockModel},
-    chainInfos: Map<string, IBlockInfoExtended>[],
+    notYetValidatedBlocks: Map<string, IBlockInfoExtended>,
     startingBlockHeight: number,
     heightOfEarliestKnownValidBlock: number) {
 
     for (let blockHash in blockData) {
       const block = blockData[blockHash];
 
       if (block.height >= startingBlockHeight && block.height <= heightOfEarliestKnownValidBlock) {
-        const indexInChainInfos = block.height - startingBlockHeight;
-        BitcoinProcessor.addBlockToChainInfo(chainInfos, indexInChainInfos, block);
+        const blockReward = BitcoinProcessor.getBitcoinBlockReward(block.height);
+        notYetValidatedBlocks.set(
+          block.hash,
+          {
+            height: block.height,
+            hash: block.hash,
+            totalFee: block.transactions[0].outputs[0].satoshis - blockReward,
+            transactionCount: block.transactions.length - 1, // minus one because of coinbase
+            previousHash: block.previousHash }
+          );
         await this.processSidetreeTransactionsInBlock(block);
       }
     }
   }
 
   /**
-   * Checks in the chain info to find the earliest valid block and its hash,
-   * and delete all transactions from invalid blocks in DB.
+   * Find all hashes in the notYetValidatedBlocks that are actually valid,
+   * add them to the validated list and delete them from the map.
    */
-  private async findEarliestValidBlock (
-    chainInfos: Map<string, IBlockInfoExtended>[],
+  private async findEarliestValidBlockAndAddToValidBlocks (
+    validatedBlocks: (IBlockInfoExtended | undefined)[],
```

Comment:
Discussion: what about just `validatedBlocks: IBlockInfoExtended[]`?

I know it maybe probably be slightly less quick ('coz we are trusting the runtime to do memory management), but the gain is slightly less code.

Reply:
So they are always defined and we do array.shift()?

Reply:
No no, just `validatedBlocks.push()`, I can reverse it if I need to.